### PR TITLE
openshift_checks: Fix incorrect list cast

### LINKED ISSUE #5610: docker_image_availability check fails, repo url divided into single character urls
#### Description

Looks to be a tokenizing error in the `docker_image_availability` check
```
Failure summary:


  1. Hosts:    infra.lab.variantweb.net, master.lab.variantweb.net, node1.lab.variantweb.net
     Play:     Verify Requirements
     Task:     openshift_health_check
     Message:  One or more checks failed
     Details:  check "docker_image_availability":
               One or more required Docker images are not available:
                   openshift3/ose-deployer:v3.7.0-0.127.0,
                   openshift3/ose-docker-registry:v3.7.0-0.127.0,
                   openshift3/ose-haproxy-router:v3.7.0-0.127.0,
                   openshift3/ose-pod:v3.7.0-0.127.0
               Configured registries: r (unreachable), e (unreachable), g (unreachable), i (unreachable), s (unreachable), t (unreachable), r (unreachable), y (unreachable), . (unreachable), o (unreachable), p (unreachable), s (unreachable), . (unreachable), o (unreachable), p (unreachable), e (unreachable), n (unreachable), s (unreachable), h (unreachable), i (unreachable), f (unreachable), t (unreachable), . (unreachable), c (unreachable), o (unreachable), m (unreachable), registry.access.redhat.com
               Checked by: timeout 10 skopeo inspect --tls-verify=false docker://{registry}/{image}
```

##### Version

```
$ ansible --version
ansible 2.3.2.0
  config file = /home/sjennings/projects/go/src/github.com/openshift/openshift-ansible/ansible.cfg
  configured module search path = Default w/o overrides
  python version = 2.7.13 (default, Jun 26 2017, 10:20:05) [GCC 7.1.1 20170622 (Red Hat 7.1.1-3)]

$ git describe
openshift-ansible-3.7.0-0.135.0-10-g62cb2a8d5
```

##### Steps To Reproduce
1. run playbook/byo/config.yml
2. playbook fails with above error

##### Expected Results
Installation check succeeds and each character of the `openshift_docker_additional_registries` is not treated as an individual repo url.

@sdodson

### PR DESCRIPTION
**Pull request by ashcrow**
``docker_image_availability`` casted ``openshift_docker_additional_registries``
to a list using the ``list()`` function. If a string was returned (IE: only
a single registry added) the result would be the string split up by
component characters. This change forces a string result from ``get_var`` to
be placed inside a list. If the result is anything BUT a string the
original ``list()`` function is called on the result.

Fixes: https://bugzilla.redhat.com/show_bug.cgi?id=1497274

Resolves #5610

### CODE REVIEW

**Comment by ashcrow** [comment]
I don't see how this is related. Is a rebase on master needed to pull in a change?

```
Oct 02 16:23:14 ocp-master origin-master-api[8864]: Error response from daemon: No such container: origin-master-api
```

**Comment by ashcrow** [comment]
bot, retest this please

**Comment by sosiouxme** [comment]
@ashcrow I believe this is the change that introduced the problem: https://github.com/openshift/openshift-ansible/pull/5559

I'm guessing our CI doesn't test with additional registries; or maybe it does, and the check just passes because the images are available locally or from the default registry.

**Comment by ashcrow** [comment]
Atomic failure is not related to this change.

```
  1. Hosts:    ocp-master
     Play:     Configure containerized nodes
     Task:     restart node
     Message:  Unable to start service origin-node: Job for origin-node.service failed because the control process exited with error code.
               See "systemctl  status origin-node.service" and "journalctl  -xe" for details.
```

File: roles/openshift_health_checker/openshift_checks/docker_image_availability.py [hunk 1]
```diff
@@ -153,7 +154,15 @@ def is_image_local(self, image):
 
     def known_docker_registries(self):
         """Build a list of docker registries available according to inventory vars."""
-        regs = list(self.get_var("openshift_docker_additional_registries", default=[]))
+        regs = self.get_var("openshift_docker_additional_registries", default=[])
+        # https://bugzilla.redhat.com/show_bug.cgi?id=1497274
+        # if the result was a string type, place it into a list. We must do this
+        # as using list() on a string will split the string into it's characters.
```
Comment (sosiouxme) [inline roles/openshift_health_checker/openshift_checks/docker_image_availability.py]: "its" :)

Not worth holding up the PR over.

Comment (ashcrow) [reply roles/openshift_health_checker/openshift_checks/docker_image_availability.py]: I'll fix it anyway 😆

**Comment by sosiouxme** [comment]
/lgtm
I don't really like that this duplicates the logic elsewhere for determining this value. It's pretty simple logic though so I think we can live with it until it's time to refactor all the role variables properly that feed into checks.

**Comment by ashcrow** [comment]
@sosiouxme agreed

**Comment by ashcrow** [comment]
flake https://github.com/openshift/origin/issues/16246

File: roles/openshift_health_checker/openshift_checks/docker_image_availability.py (hunk 1 above)
Comment (michaelgugino) [reply roles/openshift_health_checker/openshift_checks/docker_image_availability.py]: Need to check for openshift_deployment_type == 'openshift-enterprise' and addend the enterprise image registry.

**Pull request merged**